mods/default/functions.lua

```diff
@@ -481,3 +481,35 @@ minetest.register_abm({
 		end
 	end
 })
+
+
+--
+-- Checks if specified volume intersects a protected volume
+--
+
+-- Checks corners first then interior volume at specified intervals
+
+function default.intersects_protection(minp, maxp, name, interval)
+	if minetest.is_protected({x = minp.x, y = minp.y, z = minp.z}, name) or
+			minetest.is_protected({x = minp.x, y = minp.y, z = maxp.z}, name) or
+			minetest.is_protected({x = minp.x, y = maxp.y, z = minp.z}, name) or
+			minetest.is_protected({x = minp.x, y = maxp.y, z = maxp.z}, name) or
+			minetest.is_protected({x = maxp.x, y = minp.y, z = minp.z}, name) or
+			minetest.is_protected({x = maxp.x, y = minp.y, z = maxp.z}, name) or
+			minetest.is_protected({x = maxp.x, y = maxp.y, z = minp.z}, name) or
+			minetest.is_protected({x = maxp.x, y = maxp.y, z = maxp.z}, name) then
+		return true
+	end
+
+	for z = minp.z + interval, maxp.z - 1, interval do
+	for y = minp.y + interval, maxp.y - 1, interval do
+	for x = minp.x + interval, maxp.x - 1, interval do
+		if minetest.is_protected({x = x, y = y, z = z}, name) then
```

Comment:
It looks like a slightly modified form of my example above is still possible. Assuming x increases from left to right:
```
LLLLLL|LL        (protected area on this side)
^--------------- minp.x                                 checked: not protected
     ^---------- minp.x + 1 * interval                  checked: not protected
          ^----- minp.x + 2 * interval > maxp.x         not checked
      ^^^------- griefed
```
Using interval checking, it will never be possible to catch everything, but I think this is a case that ought to be caught. Else every protected area would have to be made higher than any tree could be (which apparently is about 55 nodes for moretrees atm)